### PR TITLE
Devise gem installed and setted up successfully

### PR DESCRIPTION
**Pull request by alanlamaswolox**
## Summary
I installed and configured the devise gem, and created a new user in the postgres database

## Trello
https://trello.com/c/dpwjRwEH/9-9-instalar-configurar-devise-y-crear-user

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -23,9 +23,11 @@ default: &default
 
 development:
   <<: *default
-  database: kick_development
+  #database: kick_development
```
Comment (fedeest) [inline config/database.yml]: ?

Comment (fedeest) [reply config/database.yml]: You have the database name key repeated many times

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (fedeest) [inline app/models/user.rb]: You should erase the ones that you don't use.

In this case, you only need recoverable and rememberable

**Review by fedeest — CHANGES_REQUESTED** [review]

File: Gemfile [hunk 3]
```diff
@@ -29,6 +29,7 @@ gem 'jbuilder', '~> 2.5'
 # gem 'redis', '~> 4.0'
 # Use ActiveModel has_secure_password
 # gem 'bcrypt', '~> 3.1.7'
+gem 'devise'
```
Comment (nicozar95) [inline Gemfile]: Please lock the version

File: db/migrate/20180911160324_devise_create_users.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class DeviseCreateUsers < ActiveRecord::Migration[5.1]
+  def change
+    create_table :users do |t|
+      ## Database authenticatable
+      t.string :email,              null: false, default: ""
+      t.string :encrypted_password, null: false, default: ""
+      t.string :first_name, null: false, default: ""
+      t.string :last_name, null: false, default: ""
```
Comment (nicozar95) [inline db/migrate/20180911160324_devise_create_users.rb]: Remove the default for `first_name` and `last_name`

**Review by nicozar95 — CHANGES_REQUESTED** [review]
Lock the devise version, remove the default on the migration and check on fede's review

File: db/migrate/20180911160324_devise_create_users.rb [hunk 5]
```diff
@@ -6,8 +6,8 @@ def change
       ## Database authenticatable
       t.string :email,              null: false, default: ""
       t.string :encrypted_password, null: false, default: ""
-      t.string :first_name, null: false, default: ""
-      t.string :last_name, null: false, default: ""
+      t.string :first_name, null: false
+      t.string :last_name, null: false
```
Comment (fedeest) [inline db/migrate/20180911160324_devise_create_users.rb]: Oh, i forgot: if you add this here (null: false) you should also add a validation to the model the checks for this.

Check this: https://guides.rubyonrails.org/active_record_validations.html The validation that checks for `presence true`

**Review by fedeest — CHANGES_REQUESTED** [review]

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :recoverable, :rememberable, :database_authenticatable
+  validates :email, presence: true
```
Comment (nicozar95) [inline app/models/user.rb]: add a new line between devise and validates

File: app/models/user.rb [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+  validates :encrypted_password, presence: true
```
Comment (nicozar95) [inline app/models/user.rb]: Remove this validation, devise already does it

**Review by nicozar95 — CHANGES_REQUESTED** [review]
Just the last changes

**Review by nicozar95 — APPROVED** [review]
Greate work!!! :+1: :+1:

**Review by fedeest — APPROVED** [review]

**Pull request merged**